src/qttools/utils/mpi_utils.py

```diff
@@ -2,10 +2,12 @@
 
 from pathlib import Path
 
-import numpy as np
+import scipy.sparse as sps
```

Comment:
Why do you still need scipy.sparse explicitly here? And don't use the sparse from your __init__?

Reply:
EDIT: I guess it's to use the load_npz from host side, why not using: https://docs.cupy.dev/en/latest/reference/generated/cupy.load.html in the case of xp = cp?